docs/static/spec/openapi/logstash-api.yaml

```diff
@@ -391,4 +893,371 @@ components:
+        young:
+          $ref: "#/components/schemas/JVMMemoryPool"
+    JVMGCCollector:
+      type: object
+      properties:
+        collection_time_in_millis:
+          type: integer
+          format: int64
+        collection_count:
+          type: integer
+    JVMGCCollectors:
+      type: object
+      properties:
+        old:
+          $ref: "#/components/schemas/JVMGCCollector"
+        young:
+          $ref: "#/components/schemas/JVMGCCollector"
+
+    ProcessStats:
+      type: object
+      properties:
+        process:
+          type: object
+          properties:
+            open_file_descriptors:
+              type: integer
+            peak_open_file_descriptors:
+              type: integer
+            max_file_descriptors:
+              type: integer
+            mem:
+              type: object
+              properties:
+                total_virtual_in_bytes:
+                  type: integer
+            cpu:
+              type: object
+              properties:
+                total_in_millis:
+                  type: integer
+                percent:
+                  type: integer
+                load_average:
+                  type: object
+                  properties:
+                    "1m":
+                      type: number
+
+    ThroughputStats:
+      type: object
+      properties:
+        throughput:
+          $ref: "#/components/schemas/FlowWindows"
+    WorkerStats:
+      type: object
+      properties:
+        worker_utilization:
+          $ref: "#/components/schemas/FlowWindows"
+        worker_millis_per_event:
+          $ref: "#/components/schemas/FlowWindows"
+
+    InputPluginStats:
+      type: object
+      properties:
+        id:
+          type: string
+        name:
+          type: string
+        flow:
+          $ref: "#/components/schemas/ThroughputStats"
+        events:
+          $ref: "#/components/schemas/InputPluginEventsStats"
+    FilterOutputPluginStats:
+      type: object
+      properties:
+        id:
+          type: string
+        name:
+          type: string
+        flow:
+          $ref: "#/components/schemas/WorkerStats"
+        events:
+          $ref: "#/components/schemas/FilterOutputPluginEventsStats"
+    CodecStats:
+      type: object
+      properties:
+        id:
+          type: string
+        name:
+          type: string
+        decode:
+          type: object
+          properties:
+            duration_in_millis:
+              type: integer
+            writes_in:
+              type: integer
+            out:
+              type: integer
+        encode:
+          type: object
+          properties:
+            duration_in_millis:
+              type: integer
+            writes_in:
+              type: integer
+
+    PipelineStats:
+      type: object
+      properties:
+        pipelines:
+          type: object
+          additionalProperties:
+            description: "pipeline name"
```

Comment:
I don't think that description makes sense in this context, so IMO it either needs to be improved or removed.
```suggestion

```

Reply:
In OpenAPI, `additionalProperties` means a map structure where the key / field name is dynamic generated 
{ 
  additionalProperty1: { ... }
  additionalProperty2: { ... }
}
Description is to tell user the dynamic part `additionalProperty1` is a pipeline name, where the pipeline name is a user defined value.


Take a static field example from `_node/os`
{
  "os": {
    "name": "Mac OS X",
    "arch": "x86_64",
    "version": "10.12.4",
    "available_processors": 8
  }
}
os, name, arch, version, available_processors never change

Reply:
updated the description to explain the key and value